### PR TITLE
[RDY] Simplify infection condition checking code, re-orgnize 'and', rename fee variable

### PR DESCRIPTION
**Pull request by Alberth289346**

### CODE REVIEW

**Comment by Alberth289346** [comment]
@sadger  could you review this please?

Line 143 in the old version translates to lines 147 and 148 in the new file. The condition looks a bit weird to me, maybe I made an error?

Alternatively, you may have intended
````
(patient.disease == other.disease or other.disease.contagious) and not other.diagnosed
````
instead?

File: CorsixTH/Lua/epidemic.lua [hunk 1]
```diff
@@ -234,25 +239,22 @@ end
...
+  if self.result_determined then return end
 
-  if infected_patient_left() and not self.result_determined then
-    self.result_determined = true
-    if not self.inspector then
-      self:spawnInspector()
+  -- Check whether a patient has left.
+  for _, infected_patient in ipairs(self.infected_patients) do
+    local px, py = infected_patient.tile_x, infected_patient.tile_y
+    -- If leaving and longer in the hospital
```
Comment (sadger) [inline CorsixTH/Lua/epidemic.lua]: This is a typo in my original one should read "-- If leaving and **no ** longer in the hospital"

**Comment by sadger** [comment]
Just a small reminder you have a typo in your commit message, should be "re-organise" (GB) or "re-organize" (US) depending on preference.

File: CorsixTH/Lua/epidemic.lua [hunk 2]
```diff
@@ -131,24 +131,29 @@ end
...
+  @return true if patient can infect other, false otherwise (boolean) ]]
+  local function canInfectOther(patient, other)
+    -- Patient is not infectious.
+    if patient.cured or patient.vaccinated then return false end
+
+    -- 'other' is already infected or is going home.
+    if other.infected or other.cured or other.vaccinated then return false end
+    if other.is_emergency then return false end -- Don't interact with emergencies.
+
+    if patient.disease ~= other.disease and
```
Comment (sadger) [inline CorsixTH/Lua/epidemic.lua]: Comment: -- If the other patient has a different disease OR the other patient's disease cannot be changed

**Comment by Alberth289346** [comment]
fixed old comment, fixed commit message, and added explanation, all as suggested.

**Comment by Alberth289346** [comment]
rebased

**Pull request merged**